vnext/Microsoft.ReactNative/Modules/DeviceInfoModule.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "DeviceInfoModule.h"
+#include <IReactDispatcher.h>
+#include <winrt/Windows.UI.Core.h>
+#include <winrt/Windows.UI.ViewManagement.h>
+
+namespace Microsoft::ReactNative {
+
+DeviceInfoHolder::DeviceInfoHolder() {
+  updateDeviceInfo();
+}
+
+void DeviceInfoHolder::InitDeviceInfoHolder(
+    const winrt::Microsoft::ReactNative::ReactPropertyBag &propertyBag) noexcept {
+  auto deviceInfoHolder = std::make_shared<DeviceInfoHolder>();
+  deviceInfoHolder->updateDeviceInfo();
+
+  propertyBag.Set(
+      winrt::Microsoft::ReactNative::ReactPropertyId<
+          winrt::Microsoft::ReactNative::ReactNonAbiValue<std::shared_ptr<DeviceInfoHolder>>>(
+          L"DeviceInfo", L"DeviceInfoHolder"),
+      std::move(deviceInfoHolder));
+
+  auto const &displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
```

Comment:
>auto const & [](start = 2, length = 12)

It is better not to use references for temporaries. It may cause crashes in Release mode. Though, I have seen somewhere that  it might work, but I have seen this issue too often.